lib/kvdb/kvdb_meta.c

```diff
@@ -285,144 +279,119 @@ check_media_class_keys(const cJSON *const node)
 }
 
 static merr_t
-parse_v1(const cJSON *const root, struct kvdb_meta *const meta, const char *const kvdb_home)
+kvdb_meta_parse(const cJSON *const root, struct kvdb_meta *const meta, const char *const kvdb_home)
```

Comment:
There are benefits to keeping distinct functions for parsing old versions instead of having one function that can parse multiple versions:
- the old parsers never need to change and by keeping them in their own function they are essentially snapshotted and less likely to be broken over time.
- it's much easier to drop support for older versions.

Reply:
I agree with the benefits that you mention and it weighs above having duplicated code, will change this.